File: TodoManager.Backend/Models/Todo.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TodoManager.Backend.Models
{
    public class Todo
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public string Id { get; set; } = Guid.NewGuid().ToString();

        public DateTime CreatedAt { get; set; } = DateTime.Now;

        public string Text { get; set; } = string.Empty;
    }
}

File: TodoManager.Frontend/src/app/todo.ts
export class Todo {
    id: string = '';
    text: string = '';
    createdAt: Date = new Date();

    copyFrom(t: ITodo){
        this.id = t.id;
        this.text = t.text;
        this.createdAt = new Date(t.createdAt);
    }
}

export interface ITodo{
    id: string;
    text: string;
    createdAt: string;
}
